### PR TITLE
Remove all files outside the source directory from the coverage report

### PR DESCRIPTION
**Pull request by helaan**
Cuurently, a set of directories is filtered out from the output based on
the location of system includes on most common linux distro's. This
commit does away with the blacklist and implements a whitelist approach:
only the files inside the source root are shown.

### CODE REVIEW

File: mesonbuild/scripts/coverage.py [hunk 1]
```diff
@@ -56,10 +52,11 @@ def coverage(source_root, build_root, log_dir):
                                '-a', initial_tracefile,
                                '-a', run_tracefile,
                                '-o', covinfo])
-        remove_dir_from_trace(lcov_exe, covinfo, '/usr/include/*')
-        remove_dir_from_trace(lcov_exe, covinfo, '/usr/local/include/*')
-        remove_dir_from_trace(lcov_exe, covinfo, '/usr/src/*')
-        remove_dir_from_trace(lcov_exe, covinfo, '/usr/lib/llvm-*/include/*')
+        # Remove all directories outside the source_root from the covinfo
+        subprocess.check_call([lcov_exe,
+                               '--extract', raw_tracefile,
```
Comment (ignatenkobrain) [inline mesonbuild/scripts/coverage.py]: __[Flake8]__

[W291] trailing whitespace

Comment (helaan) [reply mesonbuild/scripts/coverage.py]: Whoops. First time contributor, should I amend my commit and force push a new version to my fork?

Comment (QuLogic) [reply mesonbuild/scripts/coverage.py]: Yes.

**Comment by jpakkane** [comment]
Thanks.

**Pull request merged**